crates/proof-of-sql/src/base/math/decimal.rs

```diff
@@ -1,10 +1,37 @@
 //! Module for parsing an `IntermediateDecimal` into a `Decimal75`.
 use crate::{
-    base::scalar::Scalar,
-    sql::parse::{ConversionError, ConversionResult},
+    base::{
+        math::decimal::DecimalError::{PrecisionParseError, RoundingError},
+        scalar::Scalar,
+    },
+    sql::parse::{
+        ConversionError::{self, DecimalConversionError},
+        ConversionResult,
+    },
 };
 use proof_of_sql_parser::intermediate_decimal::IntermediateDecimal;
 use serde::{Deserialize, Deserializer, Serialize};
+use thiserror::Error;
+
+/// Errors related to decimal operations.
+#[derive(Error, Debug, Eq, PartialEq)]
+pub enum DecimalError {
+    #[error("Invalid decimal format or value: {0}")]
+    /// Error when a decimal format or value is incorrect
+    InvalidDecimal(String),
+
+    #[error("Decimal precision is not valid: {0}")]
+    /// Decimal precision exceeds the allowed limit
+    InvalidPrecision(String),
+
+    #[error("Error while parsing precision from query: {0}")]
+    /// Error in parsing precision in a query
+    PrecisionParseError(String),
+
+    #[error("Unsupported operation: cannot round decimal: {0}")]
+    /// Decimal rounding is not supported
+    RoundingError(String),
+}
```

Comment:
What exactly are the use cases of each variant?
I think there are several classes of decimal errors here:
1. The string isn't even a decimal e.g. "notastring", "-21.233.122" etc aka `InvalidDecimal`
2. Invalid precision e.g. precision above 75/76/whatever set by `Scalar` or non-positive aka `InvalidPrecision`
3. Invalid scale e.g. operations that cause scale to be outside i8 range aka `InvalidScale`
Unless explicit rounding happens what we consider to be `RoundingError` is in reality an `InvalidPrecision` since in order not to round the precision will hit the upper bound.